src/cmap/test/integration.rs

```diff
@@ -58,3 +63,84 @@ async fn acquire_connection_and_send_command() {
     assert!(names.iter().any(|name| name == "admin"));
     assert!(names.iter().any(|name| name == "config"));
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn concurrent_connections() {
+    let _guard = LOCK.run_exclusively().await;
+
+    let client = TestClient::new().await;
+    let version = VersionReq::parse(">= 4.2.9").unwrap();
+    // blockConnection failpoint option only added in 4.4.
+    if !version.matches(&client.server_version) {
+        println!(
+            "skipping concurrent_connections test due to server not supporting failpoint option"
+        );
+        return;
+    }
+
+    // stall creating connections for a while
+    let failpoint = doc! {
+        "configureFailPoint": "failCommand",
+        "mode": "alwaysOn",
+        "data": { "failCommands": [ "isMaster" ], "blockConnection": true, "blockTimeMS": 1000 }
+    };
+    client
+        .database("admin")
+        .run_command(failpoint, None)
+        .await
+        .expect("failpoint should succeed");
+
+    let handler = Arc::new(EventHandler::new());
+    let client_options = CLIENT_OPTIONS.clone();
+    let mut options = ConnectionPoolOptions::from_client_options(&client_options);
+    options.event_handler = Some(handler.clone() as Arc<dyn crate::cmap::CmapEventHandler>);
+    let pool = ConnectionPool::new(
+        CLIENT_OPTIONS.hosts[0].clone(),
+        Default::default(),
+        Some(options),
+    );
+
+    let mut tasks: Vec<AsyncJoinHandle<()>> = Default::default();
```

Comment:
I think you can do this using `map` on the range; just don't put `move` in front of the closure being passed in, and then `pool` will only be borrowed, allowing it to be cloned each invocation.

Reply:
yep, done.